src/main/java/org/beanpod/switchboard/controller/DecoderController.java

```diff
@@ -38,7 +42,10 @@
 
     @GetMapping
     public List<DecoderDTO> retrieveAllDecoders() {
-        return decoderMapper.toDecoderDTOs(decoderService.getDecoders());
+        //TODO this is necessary to avoid a stackoverflow since Decoders reference input channels which reference Decoders. I suggest adding CRUD endpoints to channels and not referencing the channels in the Encoder/Decoder at all.
+        List<DecoderEntity> decoderEntity = decoderService.getDecoders();
+        decoderEntity.forEach(this::removeDecoderReferences);
```

Comment:
what is the purpose of line 47, if i understand what you are trying to do, you can use 
@JsonIgnoreProperties({"hibernateLazyIntializer", "handler", "X"}), x is the attribute name that references the other entity. that will prevent the infinete refrences

Reply:
Or as you mentioned we can do a unidirectional mapping from channel to decoder/encoder.

Reply:
I wasn't able to get JsonIngoreProperties to work, I tried a whole bunch of different ways. I think it doesn't want to work because I have to ignore a nested object, for example ingoring `encoder` in `inputs` int `EncoderEntity`. Also we have different ignore requirements depending on how we retrieve those objects. When we get an `De/EncoderEntity` we want to ignore `channel.en/decoder` and when we retrieve just the channel we want to ignore `outputs`/`inputs`.

Reply:
Yes, the unidirectional mapping would be nice, but that would need an entire new controller and I didn't have time to do that.